extensions/PhabBugz/lib/Util.pm

```diff
@@ -53,28 +55,50 @@ sub get_revision_by_id {
 }
 
 sub create_revision_attachment {
-    my ($bug, $revision_id, $revision_title) = @_;
+    my ( $bug, $revision_id, $revision_title ) = @_;
+
+    my $phab_base_uri = Bugzilla->params->{phabricator_base_uri};
+    ThrowUserError('invalid_phabricator_uri') unless $phab_base_uri;
+
+    my $revision_uri = $phab_base_uri . "D" . $revision_id;
+
+    # Check for previous attachment with same revision id.
+    # If one matches then return it instead. This is fine as
+    # BMO does not contain actual diff content.
+    my $review_attachment = first {
+        $_->contenttype eq PHAB_CONTENT_TYPE
+          && trim( $_->data ) eq trim($revision_uri)
+    }
+    @{ $bug->attachments };
```

Comment:
This needs some reformatting... maybe private sub that checks against a revision?
```perl
 my $review_attachment = first { _is_phabricator_revision($_) } @{ $bug->attachments };
```